setup_env.py

```diff
@@ -0,0 +1,190 @@
+
+    mongo_env["MONGO_INITDB_ROOT_USERNAME"] = mongo_username
+    mongo_env["MONGO_INITDB_ROOT_PASSWORD"] = mongo_pass
+
+
+
+    # frontend
+    vite_backend_base_url = env.get_variable("VITE_BACKEND_API_BASE_URL", "http://localhost:8000/")
+    vite_env = env.get_variable("VITE_ENV", "dev")
+    
+    frontend_env["VITE_BACKEND_API_BASE_URL"] = vite_backend_base_url
+    frontend_env["VITE_ENV"] = vite_env
+
+    # Finished reading - save all changes
+    write_env_file(backend_env, BACKEND_ENV)
+    write_env_file(frontend_env, FRONTEND_ENV)
+    write_env_file(mongo_env, MONGO_ENV)
+    write_env_file(sessionsdb_env, SESSIONSDB_ENV)
+    print("Finished writing all env files. Exiting successfully")
+
+class EnvReader():
+    def __init__(self, *, env_files: list[Path], cli_args: dict[str, Any]) -> None:
+        self.in_production = bool(cli_args.get("prod"))
+        self.default_mode = bool(cli_args.get("default"))
+        self.cli_args: dict[str, str] = {
+            k: str(v) for k, v in cli_args.items() if v is not None
+        }
+
+        self.preexisting_env: dict[str, str] = {}
+
+        for env_file in env_files:
+            parsed_env = {
+                key: value
+                for key, value in dotenv.dotenv_values(env_file).items()
+                if value is not None
+            }
+            self.preexisting_env.update(parsed_env)
+
+    def get_variable(self, name: str, default: Optional[str] = None) -> Optional[str]:
+        # CLI arguments have highest priority. If provided, dont ask for value
+        if (value_from_cli := self.cli_args.get(name.lower())) is not None:
+            print(f"Successfully read value from args. Using {name}={value_from_cli}")
+            return value_from_cli
+
+        def read_input() -> Optional[str]:
+            if self.default_mode:
+                return None
+            try:
+                entered_line = input().strip()
+                return None if not entered_line else entered_line
+            except EOFError:
+                return None
+
+        if (default_value := self.preexisting_env.get(name, default)) is not None:
+            print(f"Enter value for {name} (press [enter] to accept default '{default_value}')")
+            entered_value = read_input()
+            return entered_value or default_value
+        else:
+            print(f"Enter value for {name} (press [enter] to skip)")
+            return read_input()
+
+def parse_cli_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description="Set up env/ folder and required env files")
+    parser.add_argument("--sessionsdb_username", type=str)
+    parser.add_argument("--sessionsdb_password", type=str)
+    parser.add_argument("--sessionsdb_service_hostname", type=str)
+
+    parser.add_argument("--auth_cse_client_id", type=str)
+    parser.add_argument("--auth_cse_client_secret", type=str)
+    parser.add_argument("--auth_redirect_base_uri", type=str)
+
+    
```

Comment:
mongo args 🥺

Reply:
Added only the arg for the hostname, to make sure that the mongo and backend's user and pass args stay consistent